pruntime/enclave/src/lib.rs

```diff
@@ -1123,28 +1141,31 @@ fn query(q: types::SignedQuery) -> Result<Value, Value> {
         None => None
     };
     // Dispatch
-    let mut state = STATE.lock().unwrap();
-    let res = match opaque_query.contract_id {
-        DATA_PLAZA => serde_json::to_value(
-            state.contract1.handle_query(
-                accid_origin.as_ref(),
-                types::deopaque_query(opaque_query)
-                    .map_err(|_| error_msg("Malformed request (data_plaza::Request)"))?.request)
-        ).unwrap(),
-        BALANCE => serde_json::to_value(
-            state.contract2.handle_query(
-                accid_origin.as_ref(),
-                types::deopaque_query(opaque_query)
-                    .map_err(|_| error_msg("Malformed request (balance::Request)"))?.request)
-        ).unwrap(),
-        ASSETS => serde_json::to_value(
-            state.contract3.handle_query(
-                accid_origin.as_ref(),
-                types::deopaque_query(opaque_query)
-                    .map_err(|_| error_msg("Malformed request (assets::Request)"))?.request)
-        ).unwrap(),
-        _ => return Err(Value::Null)
-    };
+	let mut res = handle_query_receipt(opaque_query.clone(), accid_origin.clone().unwrap());
```

Comment:
We can assign a special contract_id for the receipt queries (e.g. 0). By this way there's no need to parse the entire json in order to detect if it's a Receipt query